helix-term/src/commands/typed.rs

```diff
@@ -720,8 +739,19 @@ pub fn write_all_impl(
         cx.editor.save::<PathBuf>(id, None, force)?;
     }
 
-    if !errors.is_empty() && !force {
-        bail!("{:?}", errors);
+    if force {
+        return Ok(());
+    }
+
+    let mut error_msg = errors.join(" ");
+    if !readonly_errors.is_empty() {
+        // statusline text will have the form:
+        // [/home/user/readonly.txt is readonly, ..., /readonly2.txt is readonly] add ! to overwrite
+        error_msg = format!("{:?} use ! to overwrite", readonly_errors).replace('"', "");
```

Comment:
Similar here: adding `!` won't make a difference for read-only files. I don't think we need extra handling for these errors either: the errors from the writes failing will already show up in the statusline. Improving the display of that when multiple writes fail could be nice but that could be a separate PR since writes can fail for other reasons than being read-only.